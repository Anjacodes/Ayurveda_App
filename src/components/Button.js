/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';

export const Button = ({ handleClick, addClass, text }) => (
  <button onClick={handleClick} className={addClass} type="button">{text}</button>
);

// PropTypes
Button.propTypes = {
  addClass: PropTypes.string,
  text: PropTypes.string,
  handleClick: PropTypes.func,
};

Button.defaultProps = {
  addClass: 'btn',
  text: '0',
};

export default Button;
