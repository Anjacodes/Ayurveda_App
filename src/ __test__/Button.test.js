import renderer from 'react-test-renderer';
import Button from '../components/Button';

it('Button renders correctlty', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
