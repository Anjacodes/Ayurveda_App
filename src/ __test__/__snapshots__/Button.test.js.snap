// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctlty 1`] = `
<button
  className="btn"
  type="button"
>
  0
</button>
`;
